### PR TITLE
Adding Hello Wolfi workshop kit

### PR DESCRIPTION
**Pull request by erikaheidi**
This PR brings the Wolfi 101 workshop kit to Academy. I have updated this PR to bring only the workshop kit change and not mix up with the workflow update for the autodocs github actions.

Workshop Kit preview: https://deploy-preview-243--ornate-narwhal-088216.netlify.app/open-source/wolfi/hello-wolfi/

### CODE REVIEW

File: content/open-source/wolfi/hello-wolfi.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+1. Download the `cgr.dev/chainguard/melange` and `cgr.dev/chainguard/apko` images with `docker pull`
+2. Generate melange signing keys
+3. Build the `melange-php.yaml` package
+4. Build the `composer-php.yaml` package
+5. Build the `melange-app.yaml` package
+6. Build the `apko.yaml` container image
+7. Load the image with `docker load`
+8. Run the image with `docker run --rm <image-name>`
+
+### 1. Download melange and apko images
```
Comment (ltagliaferri) [inline content/open-source/wolfi/hello-wolfi.md:52]: Would you be against adding a `git clone https://github.com/chainguard-dev/hello-wolfi-demo` and `cd hello-wolfi-demo` step before launching in here?

**Review by ltagliaferri — CHANGES_REQUESTED** [review]
I can't get this to work on macos but I can run through no problem on a linux VM. I don't think this should be a blocker but it's not great that it's sticking for a widely used OS.

Running through today, at this command:

```
docker run --privileged --rm -v "${PWD}":/work -- \
  cgr.dev/chainguard/melange build melange-php.yaml \
  --arch x86_64 \
  --signing-key melange.rsa --keyring-append melange.rsa.pub
```

I get this error:

```
2023/01/05 00:19:37 melange (php/x86_64): cc: fatal error: Killed signal terminated program cc1
2023/01/05 00:19:37 melange (php/x86_64): compilation terminated.
2023/01/05 00:19:37 melange (php/x86_64): make: *** [Makefile:863: ext/fileinfo/libmagic/apprentice.lo] Error 1
2023/01/05 00:19:37 melange (php/x86_64): make: *** Waiting for unfinished jobs....
2023/01/05 00:19:59 melange (php/x86_64): make: Leaving directory '/home/build'
2023/01/05 00:19:59 ERROR: failed to build package. the build environment has been preserved:
2023/01/05 00:19:59 melange (php/x86_64):   workspace dir: /tmp/melange-workspace-46350970
2023/01/05 00:19:59 melange (php/x86_64):   guest dir: /tmp/melange-guest-1464208290
Error: failed to build package: unable to run pipeline: exit status 2
2023/01/05 00:19:59 error during command execution: failed to build package: unable to run pipeline: exit status 2
```

When everyone is back maybe we can talk through this one.

File: content/open-source/wolfi/hello-wolfi.md (hunk 1 above)
Comment (erikaheidi) [reply content/open-source/wolfi/hello-wolfi.md:52]: no problem, I will add that!

**Comment by erikaheidi** [comment]
> I can't get this to work on macos but I can run through no problem on a linux VM. I don't think this should be a blocker but it's not great that it's sticking for a widely used OS.
> 
> Running through today, at this command:
> 
> ```
> docker run --privileged --rm -v "${PWD}":/work -- \
>   cgr.dev/chainguard/melange build melange-php.yaml \
>   --arch x86_64 \
>   --signing-key melange.rsa --keyring-append melange.rsa.pub
> ```
> 
> I get this error:
> 
> ```
> 2023/01/05 00:19:37 melange (php/x86_64): cc: fatal error: Killed signal terminated program cc1
> 2023/01/05 00:19:37 melange (php/x86_64): compilation terminated.
> 2023/01/05 00:19:37 melange (php/x86_64): make: *** [Makefile:863: ext/fileinfo/libmagic/apprentice.lo] Error 1
> 2023/01/05 00:19:37 melange (php/x86_64): make: *** Waiting for unfinished jobs....
> 2023/01/05 00:19:59 melange (php/x86_64): make: Leaving directory '/home/build'
> 2023/01/05 00:19:59 ERROR: failed to build package. the build environment has been preserved:
> 2023/01/05 00:19:59 melange (php/x86_64):   workspace dir: /tmp/melange-workspace-46350970
> 2023/01/05 00:19:59 melange (php/x86_64):   guest dir: /tmp/melange-guest-1464208290
> Error: failed to build package: unable to run pipeline: exit status 2
> 2023/01/05 00:19:59 error during command execution: failed to build package: unable to run pipeline: exit status 2
> ```
> 
> When everyone is back maybe we can talk through this one.

I'm figuring this out now, will update asap once I get some feedback from the images team.

File: content/open-source/wolfi/hello-wolfi.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+### 1. Clone the demo repository
+Start by cloning the demo repository to your local machine. This will download the demo files to your home folder, under a `hello-wolfi-demo` folder:
+
+```shell
+cd ~
+git clone https://github.com/chainguard-dev/hello-wolfi-demo.git
+cd hello-wolfi-demo
+```
+
+### 2. Set up QEMU headers to enable multi-architecture builds via Docker
```
Comment (ltagliaferri) [inline content/open-source/wolfi/hello-wolfi.md:63]: I think we can remove this section and just point to the troubleshooting guide if there is any problem

**Review by ltagliaferri — APPROVED** [review]
For this to work with macOS, I had to replace commands to use `linux/aarch64` or `--arch aarch64` in place of the mentions of `x86_64`. 

Do you want to try to make this as the first multi-version article, with one for Linux and one for macOS? Unless you think we should have both commands in the same tutorial

```
docker run --platform linux/aarch64 --privileged --rm -v "${PWD}":/work --   cgr.dev/chainguard/melange build melange-php.yaml   --arch aarch64   --signing-key melange.rsa --keyring-append melange.rsa.pub
```

File: content/open-source/wolfi/hello-wolfi.md (hunk 2 above)
Comment (ltagliaferri) [reply content/open-source/wolfi/hello-wolfi.md:63]: (this did not solve the macos issue)

Comment (erikaheidi) [reply content/open-source/wolfi/hello-wolfi.md:63]: alright!

**Comment by erikaheidi** [comment]
@ltagliaferri I almost forgot about this one, do you want me to add a separate section with the macOS commands? you could share the code block that worked for you

File: content/open-source/wolfi/hello-wolfi.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+### 2. Generate melange signing keys
+To make sure the generated packages work with apko, you'll need to sign them. The following command will generate a keypair that you can use when building your packages:
+
+```shell
+docker run --rm -v "${PWD}":/work cgr.dev/chainguard/melange keygen
+```
+
+### 3. Build the PHP package
+Next, build the PHP package with melange.
+
```
Comment (ltagliaferri) [inline content/open-source/wolfi/hello-wolfi.md:79]:
```suggestion
On a Linux machine, follow the next command.
```

File: content/open-source/wolfi/hello-wolfi.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+```shell
+docker run --privileged --rm -v "${PWD}":/work -- \
+  cgr.dev/chainguard/melange build melange-php.yaml \
+  --arch x86_64 \
+  --signing-key melange.rsa --keyring-append melange.rsa.pub
+```
+
```
Comment (ltagliaferri) [inline content/open-source/wolfi/hello-wolfi.md:86]:
```suggestion
On a macOS machine, follow the next command.
  
```shell
docker run --privileged --rm -v "${PWD}":/work -- \
  cgr.dev/chainguard/melange build melange-php.yaml \
  --arch aarch64 \
  --signing-key melange.rsa --keyring-append melange.rsa.pub
```

File: content/open-source/wolfi/hello-wolfi.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+_If you run into issues while running melange commands, check the [melange troubleshooting guide](/open-source/melange/troubleshooting/)._
+
+
+### 4. Build the Composer package
+You can now build the Composer package with the following command:
```
Comment (ltagliaferri) [inline content/open-source/wolfi/hello-wolfi.md:91]:
```suggestion
You can now build the Composer package with the following command on Linux operating systems:
```

File: content/open-source/wolfi/hello-wolfi.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+```shell
+docker run --privileged --rm -v "${PWD}":/work -- \
+  cgr.dev/chainguard/melange build melange-composer.yaml \
+  --arch x86_64 \
+  --signing-key melange.rsa --keyring-append melange.rsa.pub
+```
+
```
Comment (ltagliaferri) [inline content/open-source/wolfi/hello-wolfi.md:99]:
```suggestion
On macOS, you can build the Composer package with the following command:

```shell
docker run --privileged --rm -v "${PWD}":/work -- \
  cgr.dev/chainguard/melange build melange-composer.yaml \
  --arch aarch64 \
  --signing-key melange.rsa --keyring-append melange.rsa.pub
```

File: content/open-source/wolfi/hello-wolfi.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+### 5. Build the app package
+With both the PHP and Composer dependencies in place, you can now build the application package with the following command:
```
Comment (ltagliaferri) [inline content/open-source/wolfi/hello-wolfi.md:101]:
```suggestion
With both the PHP and Composer dependencies in place, you can now build the application package with the following command on Linux machines:
```

File: content/open-source/wolfi/hello-wolfi.md [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+```shell
+docker run --privileged --rm -v "${PWD}":/work -- \
+  cgr.dev/chainguard/melange build melange-app.yaml \
+  --arch x86_64 \
+  --signing-key melange.rsa --keyring-append melange.rsa.pub
+```
+
```
Comment (ltagliaferri) [inline content/open-source/wolfi/hello-wolfi.md:109]: docker run --platform linux/aarch64 --privileged --rm -v "${PWD}":/work -- \ 
  cgr.dev/chainguard/melange build melange-php.yaml   --arch aarch64   --signing-key melange.rsa --keyring-append melange.rsa.pub
  
 ```suggestion
On macOS, you can now build the application package now that you have the PHP and Composer dependencies in place.

```shell
docker run --privileged --rm -v "${PWD}":/work -- \
  cgr.dev/chainguard/melange build melange-app.yaml \
  --arch aarch64 \
  --signing-key melange.rsa --keyring-append melange.rsa.pub
```

**Review by ltagliaferri — APPROVED** [review]
Thanks, @erikaheidi, I added the commands here, I think it would be super nice to have a tabular workflow like they have on the Cockroach Docs [Getting Started Guide](https://www.cockroachlabs.com/docs/cockroachcloud/quickstart.html). For now, use your best judgement about how to incorporate. I left suggestions but do what you think is best.

I also just fully tested it again with the commands I entered so we are good to go!

**Comment by erikaheidi** [comment]
> Thanks, @erikaheidi, I added the commands here, I think it would be super nice to have a tabular workflow like they have on the Cockroach Docs [Getting Started Guide](https://www.cockroachlabs.com/docs/cockroachcloud/quickstart.html). For now, use your best judgement about how to incorporate. I left suggestions but do what you think is best.
> 
> I also just fully tested it again with the commands I entered so we are good to go!

Nice. Yeah that kind of tabular view would be very helpful to us, I looked into the theme features but they don't seem to have something similar. I will open a separate issue so we don't forget to search for an option, maybe there is an extension.

**Pull request merged**